### DISCUSSION TITLE
Splitting supervised segments using alignment information

### DISCUSSION THREAD

**Discussion by robflynnyh** [lhotse-speech/lhotse#755] [Q&A]
Hi, I am working with the AMI dataset and want to split up segments that exceed a certain length (some segments in AMI are up to 80 seconds long). I altered the prepare_ami script to keep the alignment info and store it in the segments and am wondering if there is any method already implemented that will split up a supervised segment in some reasonable way based on a max duration and alignments. From what I can see there isn't although I wanted to check before I write something.

**Comment by pzelasko** [comment]
Not really, the SupervisionSegments don't support splitting out of the box because alignment is an optional attribute (and most often unavailable). If you have alignments it should be fairly simple to write.

**Reply by robflynnyh** [reply]
Okay, thanks for the reply.

**Comment by desh2608** [comment]
This should now be possible to do: https://github.com/lhotse-speech/lhotse/pull/1115 (e.g. `cuts_new = cuts.trim_to_alignments(max_segment_duration=10`).

**Reply by robflynnyh** [reply]
Thanks for the update
